### PR TITLE
fix: account for DSF when using wheelDeltaX and wheelDeltaY values

### LINKED ISSUE #146403: After 1.66 update scroll speed is faster
After i have done 1.66 update using MacBook Pro M1 Max trackback noticed that in code window scrolling is faster.

Version: 1.66.0
Electron: 17.2.0
Chromium: 98.0.4758.109
Node.js: 16.13.0
V8: 9.8.177.11-electron.0
OS: Darwin arm64 21.4.0

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes https://github.com/microsoft/vscode/issues/146403

### CODE REVIEW

**Comment by deepak1556** [comment]
Builds for testing https://monacotools.visualstudio.com/Monaco/_build/results?buildId=249157&view=results

File: src/vs/base/browser/ui/scrollbar/verticalScrollbar.ts [hunk 1]
```diff
@@ -49,7 +50,7 @@ export class VerticalScrollbar extends AbstractScrollbar {
 				right: undefined,
 				bgWidth: options.verticalScrollbarSize,
 				bgHeight: options.arrowSize,
-				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(null, 0, 1)),
+				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(dom.getActiveWindow(), null, 0, 1)),
```
Comment (bpasero) [inline src/vs/base/browser/ui/scrollbar/verticalScrollbar.ts:53]: I would have expected we know the window here better.

File: src/vs/base/browser/ui/scrollbar/verticalScrollbar.ts [hunk 2]
```diff
@@ -61,7 +62,7 @@ export class VerticalScrollbar extends AbstractScrollbar {
 				right: undefined,
 				bgWidth: options.verticalScrollbarSize,
 				bgHeight: options.arrowSize,
-				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(null, 0, -1)),
+				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(dom.getActiveWindow(), null, 0, -1)),
```
Comment (bpasero) [inline src/vs/base/browser/ui/scrollbar/verticalScrollbar.ts:65]: Same here.

File: src/vs/base/browser/ui/scrollbar/horizontalScrollbar.ts [hunk 3]
```diff
@@ -49,7 +50,7 @@ export class HorizontalScrollbar extends AbstractScrollbar {
 				right: undefined,
 				bgWidth: options.arrowSize,
 				bgHeight: options.horizontalScrollbarSize,
-				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(null, 1, 0)),
+				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(dom.getActiveWindow(), null, 1, 0)),
```
Comment (bpasero) [inline src/vs/base/browser/ui/scrollbar/horizontalScrollbar.ts:53]: Should be known better.

File: src/vs/base/browser/ui/scrollbar/horizontalScrollbar.ts [hunk 4]
```diff
@@ -61,7 +62,7 @@ export class HorizontalScrollbar extends AbstractScrollbar {
 				right: arrowDelta,
 				bgWidth: options.arrowSize,
 				bgHeight: options.horizontalScrollbarSize,
-				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(null, -1, 0)),
+				onActivate: () => this._host.onMouseWheel(new StandardWheelEvent(dom.getActiveWindow(), null, -1, 0)),
```
Comment (bpasero) [inline src/vs/base/browser/ui/scrollbar/horizontalScrollbar.ts:65]: Same here.

**Review by bpasero — CHANGES_REQUESTED** [review]
Cool, I have not verified this works. Only feedback I have is to avoid `getActiveWindow` if we can pass the window from the outside.

**Review by bpasero — DISMISSED** [review]

File: src/vs/base/browser/mouseEvent.ts [hunk 5]
```diff
@@ -128,7 +128,7 @@ export class StandardWheelEvent {
 	public readonly deltaX: number;
 	public readonly target: Node;
 
-	constructor(e: IMouseWheelEvent | null, deltaX: number = 0, deltaY: number = 0) {
+	constructor(targetWindow: Window, e: IMouseWheelEvent | null, deltaX: number = 0, deltaY: number = 0) {
```
Comment (alexdima) [inline src/vs/base/browser/mouseEvent.ts:131]: Maybe we can simplify the change if we get the window from the event itself:
```
const devicePixelRatio = e.view?.devicePixelRatio || 1;
```

**Review by alexdima — COMMENTED** [review]
Very nice! I see how the change became very big to pass the target window along, but I think this could be made simpler by using `e.view` which I think is always defined despite its `Window | null` type signature.

File: src/vs/base/browser/mouseEvent.ts (hunk 5 above)
Comment (deepak1556) [reply src/vs/base/browser/mouseEvent.ts]: Addressed in https://github.com/microsoft/vscode/pull/200776/commits/6bffa641e77f5f3f313a746f953b5857c3538e0b

**Review by alexdima — APPROVED** [review]
👏 Very nice! Thank you @deepak1556 !

**Comment by milosivanovic** [comment]
@deepak1556 Is there anything else needed before this wonderful fix can be merged?

**Comment by deepak1556** [comment]
Yeah I wanted to follow-up in upstream before merging the change, created CL at https://chromium-review.googlesource.com/c/chromium/src/+/5192127

**Pull request merged**